packages/dcos-launch/extra/README.md

```diff
@@ -0,0 +1,48 @@
+* Programmatic consumption: all information necessary to interact with a given deployment configuration created by `dcos-launch` is contained in the generated `cluster_info.json` file. This allows dependent CI tasks to run other tests, describe the cluster, or completely delete the cluster
+
+## Requirements
+* Linux operating system
+* SSH client installed on localhost
+
+## Commands
+### `dcos-launch create`
+Consumes a [launch config file](launch/sample_configs/README.md) file, performs basic validation on the deployment parameters, and then signals the deployment provider to begin deployment. By default, `dcos-launch` will expect a launch config at `config.yaml` but any arbitrary path can be passed with the `-c` option. If creation is triggered successfully, then a `cluster_info.json` file will be produced for use with other `dcos-launch` commands. This path is also configurable via the `-i` command line option
+
+In the case of third-party provisioning (provider is AWS or Azure), the cluster will eventually finish deploying with no further action. In the case of onprem provisioning, `dcos-launch` needs to partially drive the deployment process, so the `dcos-launch create` command only triggers creation of the underlying bare hosts, while `dcos-launch wait` will step through the DC/OS installer stages.
+
+### `dcos-launch wait`
+Reads the cluster info from the create command to see if the deployment is ready and blocks if the cluster is not ready. In the case of third-party providers, this command will query their deployment service and surface any errors that may have derailied the deployment. In the case of onprem provisioning, there are multiple, triggerable stages in deployment, so the wait command *must* be used to complete the deployment.
+
+### `dcos-launch describe`
+Reads the cluster info and outputs the essential parameters of the cluster. E.g. master IPs, agent IPs, load balancer addresses. Additionally, the STDOUT stream is formatted in JSON so that the output can be piped into another process or tools like `jq` can be used to pull out specific paramters.
+
+### `dcos-launch pytest`
+Reads the cluster info and runs the [dcos-integration-test](../../dcos-integration-test/extra) package. This is the same test suite used to validate DC/OS pull requests. Additionally, arbitrary arguments and environment variables may be added to the pytest command. For example, if one only wants to run the cluster composition test to see if the cluster is up: `dcos-launch pytest -- test_composition.py`. Anything input after `--` will be injected after `pytest` (which is naturally run without options).
+
+### `dcos-launch delete`
+Reads the cluster info and triggers the destruction of the deployment. In cases where `dcos-launch` provided third-party dependencies via a helper (`zen_helper` or `key_helper` for AWS), `delete` will block until all those resources have been removed.
```

Comment:
Anything special here about onprem? Previously, `wait` is required to complete onprem provisioning. Does `delete` block for both onprem and third-party?